asm_offset isn't defined on non-x86 architectures e.g., aarch64, powerpc64.

CMake Error at loader/CMakeLists.txt:311 (target_compile_definitions):
  Cannot specify compile definitions for target "asm_offset" which is not
  built by this project.

--- loader/CMakeLists.txt.orig	2021-11-09 19:40:47 UTC
+++ loader/CMakeLists.txt
@@ -230,6 +230,9 @@ else(UNIX AND NOT APPLE) # i.e.: Linux
         add_custom_command(OUTPUT gen_defines.asm DEPENDS asm_offset COMMAND asm_offset GAS)
         add_custom_target(loader_asm_gen_files DEPENDS gen_defines.asm)
         target_compile_definitions(asm_offset PRIVATE _XOPEN_SOURCE=500) # hush compiler warnings for readlink
+        if(CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
+            target_compile_definitions(asm_offset PRIVATE __BSD_VISIBLE=1)
+        endif()
     else()
         message(WARNING "Could not find working x86 GAS assembler\n${ASM_FAILURE_MSG}")
         set(OPT_LOADER_SRCS ${OPT_LOADER_SRCS} unknown_ext_chain.c)
@@ -308,7 +311,6 @@ else()
     target_compile_definitions(vulkan PRIVATE _GNU_SOURCE)
     if(CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
         target_compile_definitions(vulkan PRIVATE __BSD_VISIBLE=1)
-        target_compile_definitions(asm_offset PRIVATE __BSD_VISIBLE=1)
     endif()
     if(APPLE)
         find_library(COREFOUNDATION_LIBRARY NAMES CoreFoundation)
